src/fiatconnect/saga.ts

```diff
@@ -174,18 +205,24 @@ function* _getQuote({
   providerId: string
   fiatAccountType: FiatAccountType
 }) {
-  // Fetch Quote associated with the cached providerId
-  yield put(
-    fetchFiatConnectQuotes({
-      digitalAsset,
-      cryptoAmount,
-      flow,
-      providerIds: [providerId],
-    })
-  )
-  const quotes: (FiatConnectQuoteSuccess | FiatConnectQuoteError)[] = yield select(
-    fiatConnectQuotesSelector
+  const userLocation: UserLocationData = yield select(userLocationDataSelector)
+  const localCurrency: LocalCurrencyCode = yield select(getLocalCurrencyCode)
+  const fiatConnectCashInEnabled: boolean = yield select(fiatConnectCashInEnabledSelector)
+  const fiatConnectCashOutEnabled: boolean = yield select(fiatConnectCashOutEnabledSelector)
+  const fiatConnectProviders: FiatConnectProviderInfo[] | null = yield select(
+    fiatConnectProvidersSelector
   )
+  const quotes: (FiatConnectQuoteSuccess | FiatConnectQuoteError)[] = yield call(fetchQuotes, {
+    localCurrency,
+    digitalAsset,
+    cryptoAmount,
+    country: userLocation?.countryCodeAlpha2 || 'US',
+    flow,
+    fiatConnectCashInEnabled,
+    fiatConnectCashOutEnabled,
+    fiatConnectProviders:
+      fiatConnectProviders?.filter((provider) => provider.id === providerId) ?? [],
+  })
   const normalizedQuotes = normalizeFiatConnectQuotes(flow, quotes)
```

Comment:
More of a nit, but it might be worth making a helper function as it seems much of this logic is shared with `handleFetchFiatConnectQuotes` (except the providers). Otherwise the code looks pretty good!

Reply:
+1 to this. so this function was previously using `handleFetchFiatConnectQuotes` via the action right? We cannot do that now because we have a different action for refetch quote? Wonder if we can structure it so that `handleFetchFiatConnectQuotes` calls `_getQuote` and also have a separate method to select a single quote to re-use in both `handleRefetchQuote` and `handleAttemptReturnUserFlow`?